app/controllers/application_controller.rb

```diff
@@ -5,6 +5,152 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
+################ homepage ####################
+  get '/' do
+    erb :index
+  end
+  ############# signup ########################
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    @user = User.new(params)
+    @user.save
+    #binding.pry
+    session[:user_id] = @user.id
+     if logged_in?
+       redirect '/tweets'
+     else
+       redirect '/signup'
+     end
+  end
+
+  ############# login ###########################
+
+  get '/login' do
+     if logged_in?
+       redirect '/tweets'
+     else
+      erb :'users/login'
+     end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username], password: params[:password])
+    session[:user_id] = @user.id
+    if logged_in?
+      redirect '/tweets'
+    else
+      redirect '/login'
+    end
+
+  end
+
+############## logout ##################
+
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect '/login'
+    else
+      redirect '/'
+    end
+  end
+
+  ########### tweets #####################3
+
+  get '/tweets' do
+    if logged_in?
+      @user = User.find(session[:user_id])
+    erb :'tweets/tweets'
+    #binding.pry
+  else
+    redirect '/login'
+  end
+
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content].empty?
+      redirect "/tweets/new"
+    else
+    @tweet = Tweet.new(content: params[:content], user_id: session[:user_id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.